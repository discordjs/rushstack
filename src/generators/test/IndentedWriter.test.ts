// Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license.
// See LICENSE in the project root for license information.

import { IndentedWriter } from '../IndentedWriter';

test('01 Demo from docs', () => {
  const indentedWriter: IndentedWriter = new IndentedWriter();
  indentedWriter.write('begin\n');
  indentedWriter.increaseIndent();
  indentedWriter.write('one\ntwo\n');
  indentedWriter.decreaseIndent();
  indentedWriter.increaseIndent();
  indentedWriter.decreaseIndent();
  indentedWriter.write('end');

  expect(indentedWriter.toString()).toMatchSnapshot();
});

test('02 Indent something', () => {
  const indentedWriter: IndentedWriter = new IndentedWriter();
  indentedWriter.write('a');
  indentedWriter.write('b');
  indentedWriter.increaseIndent();
    indentedWriter.writeLine('c');
    indentedWriter.writeLine('d');
  indentedWriter.decreaseIndent();
  indentedWriter.writeLine('e');

  expect(indentedWriter.toString()).toMatchSnapshot();
});
