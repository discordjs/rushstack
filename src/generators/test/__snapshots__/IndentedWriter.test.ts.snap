// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`01 Demo from docs 1`] = `
"begin
  one
  two
end"
`;

exports[`02 Indent something 1`] = `
"abc
  d
e
"
`;
